File: demos/csharp-feequote/src/FeeQuote/Models/Quote.cs
namespace FeeQuote.Models;

/// <summary>One labelled line of a fee breakdown, in whole cents.</summary>
public sealed record BreakdownLine(string Label, long AmountCents);

/// <summary>
/// The priced result for a transfer. <see cref="FeeCents"/> is the sum of the
/// breakdown lines and <see cref="TotalCents"/> is the principal plus the fee,
/// which is the amount debited from the customer.
/// </summary>
public sealed record Quote(
    string TransferId,
    long FeeCents,
    long TotalCents,
    IReadOnlyList<BreakdownLine> Breakdown);

File: demos/java-feequote/src/main/java/com/feequote/models/Quote.java
package com.feequote.models;

import java.util.List;

/**
 * The fee quoted for a transfer.
 *
 * <p>{@code feeCents} is the sum of every breakdown line. {@code totalCents} is the
 * transfer amount plus {@code feeCents}, which is what the customer is debited.
 *
 * @param transferId the transfer this quote prices
 * @param feeCents the total fee in integer cents
 * @param totalCents the amount plus the fee, in integer cents
 * @param breakdown the labeled components that make up the fee
 */
public record Quote(String transferId, long feeCents, long totalCents, List<BreakdownLine> breakdown) {

    public Quote {
        breakdown = List.copyOf(breakdown);
    }
}
